lib/honeybadger/json.ex

```diff
@@ -2,15 +2,19 @@ defmodule Honeybadger.JSON do
   @moduledoc false
 
   @spec encode(term) :: {:ok, String.t()} | {:error, Jason.EncodeError.t()}
-  def encode(notice) do
-    # try to encode without going the rabbit hole
-    case safe_encode(notice) do
+  def encode(term) do
+    case safe_encode(term) do
       {:ok, output} ->
         {:ok, output}
 
+      {:error, %Jason.EncodeError{}} ->
+        term
+        |> to_encodeable()
+        |> Jason.encode()
+
```

Comment:
Is there a reason to match on both of these explicit cases? Since they have the same result (and AFAICT these are the only possible errors) would it make sense to unify them into a single clause?

Reply:
That's a fair point. Anything else should raise an exception here, so combining the cases is fine. I'll push up that change before merging.